src/roman_datamodels/datamodels/_datamodels.py

```diff
@@ -135,37 +137,59 @@ class RampModel(_RomanDataModel):
     @classmethod
     def from_science_raw(cls, model):
         """
-        Construct a RampModel from a ScienceRawModel
+        Attempt to construct a RampModel from a DataModel
+
+        If the model has a `resultantdq` attribute, this is copied into
+        the `RampModel.groupdq` attribute.
 
         Parameters
         ----------
-        model : ScienceRawModel or RampModel
-            The input science raw model (a RampModel will also work)
+        model : ScienceRawModel, TvacModel
+            The input data model (a RampModel will also work).
+
+        Returns
+        -------
+        output_model : RampModel
+            The RampModel built from the input model. If the input is already
+            a RampModel, it is simply returned.
         """
-
         if isinstance(model, cls):
             return model
+        if not isinstance(model, (ScienceRawModel, TvacModel)):
```

Comment:
Should RampModel be in this list to match the comments?

Also, I think FpsModel should be allowed in this function as well.

Reply:
RampModel is implicitly checked for in the preceding statement. However, no issue with adding here also.